src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Icu.cs

```diff
@@ -292,6 +301,15 @@ private unsafe int IndexOfOrdinalHelper(ReadOnlySpan<char> source, ReadOnlySpan<
                         throw new Exception((string)ex_result);
                     return result;
                 }
+#elif TARGET_OSX || TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
+                if (GlobalizationMode.Hybrid)
+                {
+                    NSRange result = Interop.Globalization.IndexOfNative(m_name, m_name.Length, b, target.Length, a, source.Length, options, fromBeginning);
```

Comment:
Could we have done something similar to `Interop.Globalization.IndexOf` where `matchLengthPtr` is passed as an out parameter that is intended to be set within the function itself, and then it only returns result.Location? If we can do that then we can save on not needing to define a new type `NSRange`.